### PR TITLE
Added Pymc v4 Sampler

### PR DESCRIPTION
**Pull request by PaulJonasJost**
This is to be able to use Apsara in combination with pymc. For now only a draft.

TODO:
- [x] Import for pymc via GitHub probably until official version is released
- [x] make pymc3 deprecated as it requires old versions of e.g. arviz which would need to be updated for pymc
- [x] make DensityDist work with tensor vector
- [x] rework notebooks when pymc sampler is working to fix bumpy problem
- [x] PyMC3 tests and examples have been temporarily disabled in https://github.com/ICB-DCM/pyPESTO/pull/831. They should be re-enabled here with their PyMC4 counterparts

### CODE REVIEW

**Comment by dweindl** [comment]
- [ ] PyMC3 tests and examples have been temporarily disabled in https://github.com/ICB-DCM/pyPESTO/pull/831.They should be re-enabled here with their PyMC4 counterparts

**Review by dweindl — APPROVED** [review]
I don't have any experience with pymc/aesara, but I don't see any obvious issues.
Somebody more qualified should have another look.

File: pypesto/objective/aesara.py [hunk 1]
```diff
@@ -152,6 +158,69 @@ def __deepcopy__(self, memodict=None):
...
+            ndim_supp=0,
+            ndims_params=[x_size],
+            dtype='floatX',
+        )
+
+
+@_logprob.register(AesaraObjectiveRV)
+def aesara_logprob(op, values, *inputs, **kwargs):
+    """
+    Aeppl registration of the logprob function of AesaraObjectiveRV class.
```
Comment (yannikschaelte) [inline pypesto/objective/aesara.py:216]:
```suggestion
    Apply/apple/aesarapla? registration of the logprob function of AesaraObjectiveRV class.
```

File: pypesto/sample/pymc.py [hunk 2]
```diff
@@ -123,37 +140,30 @@ def sample(self, n_samples: int, beta: float = 1.0):
             ]
 
             # convert to tensor vector
-            theta = tt.as_tensor_variable(k)
+            theta = at.as_tensor_variable(k)
 
             # use a DensityDist for the log-posterior
```
Comment (yannikschaelte) [inline pypesto/sample/pymc.py:145]: update docstring

**Review by yannikschaelte — APPROVED** [review]
looks good to me, great!

File: pypesto/objective/aesara.py (hunk 1 above)
Comment (FFroehlich) [reply pypesto/objective/aesara.py:216]: It's really called `aeppl` :D https://github.com/aesara-devs/aeppl

Comment (yannikschaelte) [reply pypesto/objective/aesara.py:216]: oh whoopsie :rofl:

**Pull request merged**